source/exe/posix/platform_checks.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <fcntl.h>
+#include <sys/resource.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include "envoy/filesystem/filesystem.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+
+static void check_sysctl_long(std::string path, long expected, Filesystem::Instance& file_system) {
+  if (file_system.fileExists(path)) {
```

Comment:
can we put these impls into a .cc file, rather than inlining them always?

Also it looks like check_sysctl_long is only used by check_platform_settings, so it should be hidden in a .cc file.